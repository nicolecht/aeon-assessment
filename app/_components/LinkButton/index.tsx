import Link from "next/link";

export default function LinkButton({
  children,
  className,
  ...props
}: React.ComponentProps<typeof Link>) {
  return (
    <Link
      className={`p-3 font-semibold text-center text-white rounded-full min-w-32 bg-primary-500 hover:bg-primary-600 ${className}`}
      {...props}
    >
      {children}
    </Link>
  );
}
