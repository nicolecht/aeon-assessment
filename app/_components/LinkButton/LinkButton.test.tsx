import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import LinkButton from ".";

describe("LinkButton component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <LinkButton href="/test">Click Me</LinkButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with the correct href", () => {
    render(<LinkButton href="/test">Click Me</LinkButton>);
    const linkElement = screen.getByRole("link", { name: /click me/i });
    expect(linkElement).toHaveAttribute("href", "/test");
  });

  it("accepts and applies additional className", () => {
    render(
      <LinkButton href="/test" className="custom-class">
        Click Me
      </LinkButton>
    );
    const linkElement = screen.getByRole("link", { name: /click me/i });
    expect(linkElement).toHaveClass("custom-class");
  });

  it("renders the provided child text", () => {
    render(<LinkButton href="/test">Click Me</LinkButton>);
    expect(screen.getByText("Click Me")).toBeInTheDocument();
  });
});
