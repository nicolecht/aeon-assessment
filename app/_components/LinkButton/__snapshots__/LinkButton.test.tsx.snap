// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="p-3 font-semibold text-center text-white rounded-full min-w-32 bg-primary-500 hover:bg-primary-600 undefined"
    href="/test"
  >
    Click Me
  </a>
</DocumentFragment>
`;
